consult.el

```diff
@@ -2429,6 +2429,38 @@ In order to select from a specific HISTORY, pass the history variable as argumen
       (delete-minibuffer-contents))
     (insert (substring-no-properties str))))
 
+;;;;; Command: consult-isearch
+
+(defun consult--isearch-read (hist)
+  (let ((str (consult--read "History: " hist :history t :sort nil)))
+     (cons str (mapconcat 'isearch-text-char-description str ""))))
+
+;;;###autoload
+(defun consult-isearch ()
+  "Read a search string with completion from history.
+
+This replaces the current search string if Isearch is active, and
+starts a new Isearch session otherwise."
+  (interactive)
+  (let* ((hist (if (if isearch-mode isearch-regexp
```

Comment:
The hist variable should go away I suppose?

Reply:
Oops...